presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/AggregationMatcher.java

```diff
@@ -82,26 +82,35 @@ public MatchResult detailMatches(PlanNode node, Session session, Metadata metada
         }
 
         for (int i = 0; i < groupingSets.size(); i++) {
-            if (!matches(groupingSets.get(i), aggregationNode.getGroupingSets().get(i))) {
+            if (!matches(groupingSets.get(i), aggregationNode.getGroupingSets().get(i), symbolAliases)) {
                 return NO_MATCH;
             }
         }
 
         return match();
     }
 
-    static <T> boolean matches(Collection<T> expected, Collection<T> actual)
+    static boolean matches(List<String> expectedAliases, List<Symbol> actualSymbols, SymbolAliases symbolAliases)
     {
-        if (expected.size() != actual.size()) {
+        if (expectedAliases.size() != actualSymbols.size()) {
             return false;
         }
 
-        for (T symbol : expected) {
-            if (!actual.contains(symbol)) {
+        int i = 0;
+        for (String alias : expectedAliases) {
+            SymbolReference reference = symbolAliases.get(alias);
+            boolean found = false;
+            while (i < actualSymbols.size()) {
+                Symbol outputSymbol = actualSymbols.get(i++);
+                if (reference.equals(outputSymbol.toSymbolReference())) {
```

Comment:
Why not convert to a Collection of actual symbols at the top of the function and use the old comparison logic?